pkg/planner/core/memtable_predicate_extractor.go

```diff
@@ -1683,16 +1798,26 @@ func (e *InfoSchemaTablesExtractor) Extract(_ PlanContext,
 	names []*types.FieldName,
 	predicates []expression.Expression,
 ) (remained []expression.Expression) {
-	var resultSet set.StringSet
+	var resultSet, resultSet1 set.StringSet
 	e.colNames = []string{"table_schema", "table_name"}
 	e.ColPredicates = make(map[string]set.StringSet)
 	remained = predicates
 	for _, colName := range e.colNames {
-		remained, e.SkipRequest, resultSet = e.extractCol(schema, names, remained, colName, true)
-		e.ColPredicates[colName] = resultSet
+		remained, e.SkipRequest, resultSet = e.extractColWithLowerOrUpper(schema, names, remained, colName)
```

Comment:
Could it be unified with the `extractCol` function? It feels like most of the code is the same.

Reply:
Yes, but I prefer to use another function since this function is hacky.